include/revng-c/mlir/Dialect/Clift/IR/CliftTypes.td

```diff
@@ -43,7 +43,8 @@ def Clift_PrimitiveType : Clift_Type<"Primitive", "primitive", [Clift_ValueType,
   let assemblyFormat = "`<`( `is_const` `=` $is_const^ `,`` `)? $kind $size `>` ";
 }
 
-def Clift_PointerType : Clift_Type<"Pointer", "pointer", [DeclareTypeInterfaceMethods<Clift_ValueType, ["getByteSize"]>, SubElementTypeInterface]> {
+
+def Clift_PointerType : Clift_Type<"Pointer", "pointer", [DeclareTypeInterfaceMethods<Clift_ValueType, ["getByteSize"]>, SubElementTypeInterface, DeclareTypeInterfaceMethods<Clift_AliasableType>]> {
```

Comment:
I don't think we agreed to make pointer types (and array types later) aliasable.

Last time we talked about this we said we want pointers and arrays to always be emitted inline, so no alias is allowed.

Did you find any reason that make this necessary, or was it just for consistency with the others?

Reply:
I originally intended to do aliases for pointers too, but ran into issues wrt. recursive types. I removed it now.